### DISCUSSION TITLE
CF 2.0 restarts after beams from Lighthouse with ID=2 are detected

### DISCUSSION THREAD

**Discussion by zboucek** [bitcraze/discussions#541] [Q&A] [answered]
CF 2.0 restarts after beams from Lighthouse with ID=2 are detected. I tested it with two version 1 basestations and four version 2 basestations. I tried it on several CF2.0 with different LH decks. For V2, unless I plug in a station with ID=2, the system works normally. After a reboot, it does not pass the assert info and reports a cf_math.c library error.

Has anyone by any chance encountered the same problem? I updated the firmware in the drones to 2022.12 using the client, but it didn't solve anything.

**Comment by krichardsson** [comment, marked as answer]
One idea is that you might have corrupt geometry or calibration data stored in the Crazyflie. I have never seen this but have sort of expected it to turn up sooner or later :-) 

At the moment we don't have an easy way of clearing the persisted data (I created a new issue for it bitcraze/crazyflie-firmware#1207)

I have a suggested workaround for now:
1. Turn on your Crazyflie and connect your client
2. In the lighthouse tab click "Load system config" and use the file (unzipped) that I have attached to this post 
3. Turn on the base station with ID=2 to see if it works (fingers crossed)

The attached file is from our flight lab and contains data for 4 base stations. Hopefully this will write valid data (but wrong for your setting) that prevents the CF from crashing. After this it should be possible to get correct calibration data and fix the geometry.

[arena-lh2-4.zip](https://github.com/bitcraze/discussions/files/10536007/arena-lh2-4.zip)

**Reply by AkashJohnSubash** [reply]
Observed same issue on CF2.1 with the 2022.12 release firmware, and 2x lighthouse 2 base stations
The log was identical to the one posted in the old [forum](https://forum.bitcraze.io/viewtopic.php?t=4901).

**Results:**
-Rebooted with both base stations off, reflashed firmware. 
 Switched on both base stations -> SYS: cfAssertNormalStart [FAIL]  

-Rebooted with both base stations off, replaced with fresh LH deck, reflashed firmware.   
 Switched on both base stations -> SYS: cfAssertNormalStart [FAIL] 

-Rebooted with base station 1 off -> SYS: cfAssertNormalStart [FAIL]  

-Rebooted with base station 2 off -> No error, but take-off not stable  

Note : Strangely my colleague's CF2.1 using the same firmware version, lighthouse deck version, base stations did not have this issue.

**my conclusion**
Possibly due to a slight movement of base station 2, after my previous calibration, leading to data corruption ?
Following instructions by "oryx" in  the old forum regarding the line of sight fixed the issue
Related crazy-flie [issue](https://github.com/bitcraze/crazyflie-firmware/issues/818)
